appgate/syncer/git.py

```diff
@@ -284,6 +395,127 @@ def github_checkout_branch(
         return generate_branch_name(), BranchOp.CREATE_AND_CHECKOUT
 
 
+class MergeRequestLike(Protocol):
```

Comment:
The idea of having a protocol is to be able to have a generic interface for all the pullrequest/mergerequest things so I would not create one protocol for each vendor.